tests/unit/s2n_session_ticket_test.c

```diff
@@ -793,63 +811,90 @@ int main(int argc, char **argv)
     {
-        EXPECT_NOT_NULL(client_config = s2n_config_new());
-        EXPECT_SUCCESS(s2n_config_set_session_tickets_onoff(client_config, 1));
-        EXPECT_SUCCESS(s2n_config_disable_x509_verification(client_config));
-        EXPECT_NOT_NULL(client_conn = s2n_connection_new(S2N_CLIENT));
-
-        EXPECT_SUCCESS(s2n_connection_set_config(client_conn, client_config));
-
-        EXPECT_NOT_NULL(server_conn = s2n_connection_new(S2N_SERVER));
-
-        EXPECT_NOT_NULL(server_config = s2n_config_new());
-        EXPECT_SUCCESS(s2n_config_set_session_tickets_onoff(server_config, 1));
-        EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(server_config, chain_and_key));
-
-        /* Create nonblocking pipes */
-        EXPECT_SUCCESS(s2n_connections_set_io_pair(client_conn, server_conn, &io_pair));
-
-        /* Set session state lifetime for 15 hours which is equal to the default lifetime of a ticket key */
-        EXPECT_SUCCESS(s2n_config_set_session_state_lifetime(server_config, S2N_SESSION_STATE_CONFIGURABLE_LIFETIME_IN_SECS));
-
-        /* Add one ST key */
-        EXPECT_SUCCESS(s2n_config_add_ticket_crypto_key(server_config, ticket_key_name1, s2n_array_len(ticket_key_name1), ticket_key1, s2n_array_len(ticket_key1), 0));
-
-        /* Add second key when the first key is very close to it's encryption peak */
-        uint64_t mock_delay = (server_config->encrypt_decrypt_key_lifetime_in_nanos / 2) - ONE_SEC_IN_NANOS;
-        EXPECT_SUCCESS(s2n_config_set_wall_clock(server_config, mock_nanoseconds_since_epoch, &mock_delay));
-        EXPECT_SUCCESS(s2n_config_add_ticket_crypto_key(server_config, ticket_key_name2, s2n_array_len(ticket_key_name2), ticket_key2, s2n_array_len(ticket_key2), 0));
-
-        /* Add third key when the second key is very close to it's encryption peak and
-         * the first key is about to transition from encrypt-decrypt state to decrypt-only state
+        const size_t iterations = 450;
+        const size_t allowed_failures = 2;
+        size_t wrong_key_selected = 0;
+
+        /* This test sets up three different ticket encryption keys at various times in their encryption lifetime. The test
+         * is meant to check that the weighted random selection algorithm correctly selects the key that is at its
+         * encryption peak. However the algorithm will sometimes pick a key that is not at its peak encryption because the
+         * selection function uses a weighted random selection algorithm. We only expect this behavior
+         * rarely so as long as the test passes the majority of the time, we consider the selection algorithm to be
+         * working as intended.
+         *
+         * Here, out of 450 iterations we allow the test to fail (2 / 450) = 0.4% of the time. This is a higher failure rate
+         * than technically needed, but it keeps us from having to do too many iterations and slowing down the test.
```

Comment:
Why 2/450 instead of 1/225? And how was that failure rate chosen-- is it related to the math for weighted key selection?